BlazarService/src/main/java/com/hubspot/blazar/listener/SlackImNotificationVisitor.java

```diff
@@ -50,15 +58,41 @@ public void visit(RepositoryBuild build) throws Exception {
       LOG.info("Not sending notifications for builds in non-terminal states or on success / cancellation");
       return;
     }
-    if (slackSession.isPresent()) {
-      LOG.info("Sending slack notification for repo build {}", build.getId().get());
-      CommitInfo commitInfo = build.getCommitInfo().get();
-      Optional<SlackUser> user = Optional.fromNullable(slackSession.get().findUserByEmail(commitInfo.getCurrent().getAuthor().getEmail()));
-      if (user.isPresent()) {
-        slackSession.get().sendMessageToUser(user.get(), "", slackUtils.buildSlackAttachment(build));
+    sendSlackMessageWithRetries(build, slackUtils.buildSlackAttachment(build), 3);
+
+  }
+
+  private void sendSlackMessageWithRetries(RepositoryBuild build, SlackAttachment attachment, int tries) {
+    tries = tries - 1;
```

Comment:
why not use something like Guava retries for all of this? Can set it to retry at varying rates based on return value or specific exceptions a bit easier